File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int start = 0;
        int end = x,mid,ans = 0;
        while(start <= end){
            mid = start + (end - start)/2;
            if(((long)mid*mid) > (long)x){
                end = mid - 1;
            }
            else if(((long)mid*mid) < (long)x){
                ans = mid;
                start = mid + 1;
            }
            if(((long)mid*mid) == x){
                ans = mid;
                break;
            }
        }
        return ans;
    }
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        start = 0
        end = x
        ans = 0
        while(start <= end):
            mid = (start + end)//2
            if(mid*mid == x):
                ans = mid
                break
            elif mid*mid > x:
                end = mid - 1
            else:
                ans = mid
                start = mid + 1
        return ans
        